curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/644a37d773b7610ac5ede66e.md

```diff
@@ -0,0 +1,182 @@
+---
+id: 644a37d773b7610ac5ede66e
+title: Step 21
+challengeType: 0
+dashedName: step-21
+---
+
+# --description--
+
+Now that you're familiar with binary numbers, it's time to finish building the function to do the conversion for you. You'll start off with a simpler solution first, then refactor that into a recursive solution.
+
+First, you need to create some arrays to store the inputs and results of the division you'll do in the following steps. These will make it easier to see how the decimal to binary conversion works.
+
+Remove the `return` statement from your `decimalToBinary` function. Then, declare variables named `inputs`, `quotients`, and `remainders`, and assign an empty array to each of them.
+
+# --hints--
+
+You should remove the `return` statement from your `decimalToBinary` function.
+
+```js
+assert.notMatch(String(decimalToBinary), /return\s*('|"|`)(\s|\d)*\1\s*;?\s*\s*\}$/);
+```
+
+You should declare variables named `inputs`, `quotients`, and `remainders`, and assign an empty array to each of them.
+
+```js
+assert.match(String(decimalToBinary), /(var|let|const)\s+inputs\s*=\s*\[\s*\]\s*;?/);
+assert.match(String(decimalToBinary), /(var|let|const)\s+quotients\s*=\s*\[\s*\]\s*;?/);
+assert.match(String(decimalToBinary), /(var|let|const)\s+remainders\s*=\s*\[\s*\]\s*;?/);
+```
```

Comment:
This should be broken up into three different tests.

Also we should mention using `const` for declaring variables since that is what is in the final solution seed code.

Reply:
added the solution using `const`